imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -0,0 +1,208 @@
+"""Decommutate HIT CCSDS data and create L1a data products."""
+
+import logging
+from collections import defaultdict
+from dataclasses import fields
+from enum import IntEnum
+
+import numpy as np
+import xarray as xr
+
+from imap_processing import decom, imap_module_directory, utils
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import write_cdf
+from imap_processing.hit import hit_cdf_attrs
+from imap_processing.hit.l0.data_classes.housekeeping import Housekeeping
+
+logger = logging.getLogger(__name__)
+
+
+class HitAPID(IntEnum):
+    """
+    HIT APID Mappings.
+
+    Attributes
+    ----------
+    HIT_AUT : int
+        Autonomy
+    HIT_HSKP: int
+        Housekeeping
+    HIT_SCIENCE : int
+        Science
+    HIT_IALRT : int
+        I-ALiRT
+    HIT_MEMDUMP : int
+        Memory dump
+    """
+
+    HIT_AUT = 1250  # Autonomy
+    HIT_HSKP = 1251  # Housekeeping
+    HIT_SCIENCE = 1252  # Science
+    HIT_IALRT = 1253  # I-ALiRT
+    HIT_MEMDUMP = 1255  # Memory dump
+
+
+def hit_l1a_data(packet_file: Path | str):
+    """
+    Process HIT L0 data into L1A data products.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+
+    Returns
+    -------
+    dict
+        List of file paths to CDF data product files
+    """
+    # Decom, sort, and group packets by apid
+    packets = decom_packets(packet_file)
+    sorted_packets = sorted(packets, key=lambda x: x.data["SHCOARSE"].derived_value)
+    grouped_data = group_data(sorted_packets)
+
+    # Create datasets
+    # TODO define keys to skip for each apid. Currently just have
+    #  a list for housekeeping.
+    skip_keys = [
+        "shcoarse",
+        "ground_sw_version",
+        "packet_file_name",
+        "ccsds_header",
+        "leak_i_raw",
```

Comment:
Don't need to make the change in this PR, but for the future if we continue to create the housekeeping CDF, we can use the builtin `dataclasses` class `InitVar` so `leak_i_raw` doesn't show up when you extract the object's field names. 

Also after SIT-3, we might want to include the CCSDS header, packet file name, and sw version in the CDF (we'll need to talk to Eric C and Grant about that), so you might only need to skip shcoarse here in the future.

Reply:
InitVar is going to be perfect for this variable. I added a TODO so we won't forget about this and updating the list of fields to skip